src/common/SearchInput.js

```diff
@@ -14,32 +16,52 @@ const styles = StyleSheet.create({
     borderWidth: 0,
     // color: BRAND_COLOR,
   },
+  icon: {
+    margin: 6,
+  }
 });
 
+type Props = {
+  onChange: (text: string) => void,
+  showCancelIcon: boolean,
+};
+
 export default class SearchInput extends Component {
 
-  props: {
-    onChange: (text: string) => void,
-  };
+  props: Props;
+  textInput: TextInput;
+
+  handleClear = () => {
+    this.textInput.clear();
+    this.props.onChange('');
+  }
 
   render() {
-    const { onChange } = this.props;
+    const { onChange, showCancelIcon } = this.props;
 
     return (
       <View style={styles.wrapper}>
         <Input
+          textInputRef={component => { this.textInput = component; }}
           style={styles.input}
           autoCorrect={false}
           enablesReturnKeyAutomatically
           selectTextOnFocus
           underlineColorAndroid="transparent"
-          clearButtonMode="always"
           autoCapitalize="none"
           placeholder="Search"
           returnKeyType="search"
           onChangeText={onChange}
           autoFocus
         />
+        { showCancelIcon &&
+          <IconCross
+            style={styles.icon}
+            size={20}
```

Comment:
The button needs to be as big as possible. Try CONTROL_SIZE from `styles.js`
Touch targets that are smaller might not work.
Suggested minimum size for iOS is 44, we might be able to get away with a bit smaller size but 20 is too small.